### ISSUE TITLE
gpu_memory_fraction for tf.contrib.lean Estimators stopped working in tensorflow 1.1.0

### ISSUE THREAD

**Issue by WolfgangGross** [tensorflow/tensorflow#9831]
I was updating to Tensorflow 1.1.0 to use the tf.estimator API and I found, that I couldn't get it to use less than the full GPU RAM for one process.
I was trying to find out where the issue is and found, that also the tf.contrib.lean.Estimators had the same problem. 

Here a code snippet to reproduce the issue:
```
import tensorflow as tf
import numpy as np
from tensorflow.contrib.learn import DNNClassifier
import os

tf.logging.set_verbosity(tf.logging.INFO)

from tensorflow.examples.tutorials.mnist import input_data
mnist = input_data.read_data_sets('MNIST_data', one_hot=True)

feature_columns = [tf.contrib.layers.real_valued_column("x", dimension=mnist.train.images.shape[1])]
model = DNNClassifier(hidden_units=[10,10],
                      feature_columns=feature_columns,
                      n_classes=10,
                      config=tf.contrib.learn.RunConfig(gpu_memory_fraction=0.1))

input_fn_train = tf.contrib.learn.io.numpy_input_fn(
                                x={"x":mnist.train.images.reshape([-1,28,28,1]).astype(np.float32)[:1000]},
                                y=np.argmax(mnist.train.labels[:1000],1).astype(np.int32),
                                batch_size=50,
                                num_epochs=50,
                                )


model.fit(input_fn=input_fn_train)

```

With tensorflow 1.0.1 it works as expected and only allocated 1/10 of the GPU RAM, but in tensorflow 1.1.0 it ignores this parameter. 

I also check with the code from the MNIST for ML beginners code  and used the GPUOptions when initializing the session:
```
gpu_options = tf.GPUOptions(per_process_gpu_memory_fraction=0.1)
sess = tf.Session(config=tf.ConfigProto(gpu_options=gpu_options))
```

It works as expected in both verions. 
Somehow this GPU options is not set correctly when using the tf.contib.learn.Estimator framework OR The gpu_memory is allocated before this call.

I asked this before on the tensorflow discuss group and I was told that is is a bug.

**Comment by ashern** [comment]
This problem affects managed_session use generally, and not only through Estimator. The per_process_gpu_memory_fraction and the allow_growth settings have no effect.

**Comment by ispirmustafa** [comment]
This should be fixed at the head. Could you please give it a try?

**Comment by ashern** [comment]
Built from head (dbb78c7783c32f8f3dc86364dc10d7d51cf61277), this does not seem to work for me. 


```
  session_config=tf.ConfigProto(
          allow_soft_placement=True,
          log_device_placement=False)
  # session_config.gpu_options.allow_growth = True
  session_config.gpu_options.per_process_gpu_memory_fraction = 0.40

  saver = tf.train.Saver(
      max_to_keep=50,
      write_version=2 ,
      sharded=True,)

  init_op = tf.global_variables_initializer()
  sv = tf.train.Supervisor(
      saver=saver,
      save_model_secs=model_params['checkpoint_interval'],
      save_summaries_secs=model_params['report_interval'],
      init_op=init_op,
      logdir=model_params['checkpoint_dir'])

  with sv.managed_session(config=session_config) as sess:

```

[1] TITAN X (Pascal) | 54'C,  22 % | 11672 / 12189 MB | asher(11117M)

The above worked previously, but please let me know if I've missed a change.

**Comment by AaronYALai** [comment]
Having the same issue here.

And I notice that "config_pb2.ConfigProto(allow_soft_placement=True)" of "monitored_session" in lines 965-975 of estimator.py has no gpu memory config
Revise the config to incorporate gpu memory fraction can manually control gpu memory usage again
    ex. config.gpu_options.per_process_gpu_memory_fraction = 0.3

**Comment by ashern** [comment]
Rebuilt from 7937fb34dbd9fd04dc878eefa6ccfe00d1262f2e and this behavior has changed. I'm not sure that it's quite correct - perhaps it's using the allow_growth option instead of per_process_gpu_memory_fraction?

[1] TITAN X (Pascal) | 62'C,  29 % | 12049 / 12189 MB | asher(457M)

is the result of:
session_config.gpu_options.per_process_gpu_memory_fraction = 0.40

Which I would expect to reserve 40% of the available memory.

**Comment by ursk** [comment]
Has this issue been fixed? Seems like `gpu_memory_fraction` still has no effect in r1.2 and r1.3

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by ursk** [comment]
This is fixed in r1.4 using the tf.Estimator, eg.
```
config = tf.ConfigProto(gpu_options=tf.GPUOptions(per_process_gpu_memory_fraction=0.5))
trainingConfig = tf.estimator.RunConfig(session_config=config)
estimator = tf.estimator.Estimator(config=trainingConfig, ...)
```
will work as desired. I don't think it's expected to be fixed in the deprecated `tf.contrib.learn.Estimator`.

**Comment by tensorflowbutler** [comment]
Nagging Assigneee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by isaprykin** [comment]
Thanks for confirming, Urs.
